app/hall.hoon

```diff
@@ -2654,15 +2654,16 @@
     ==
   ::
     ?:  =(wat ~)  &
-    %-  ~(has in wat)
```

Comment:
Is this just a style change (fair!) or did this actually break?

Reply:
It broke, because `%hasnot` isn't a valid element of the set. The other two changes to hall used `in` where they should have been using `by`, which was also caught by the typechecker.

@ohAitch I don't like messing with the type of the map inline to add a special sentinel value that gets used in one place just so that we can add a call to `+has` that will always fail.

Reply:
It "broke", in that the code was entirely dynamically valid, but nonetheless not entirely following all of the type rules.

I agree that it's ugly, it just seemed less ugly; probably the special sentinel should be named `%other` and formally in the type, if possibly never set. (`circle-data` is what the one-letter subscription modes turned into right?)

Reply:
Actually looking at the line above, it looks `%other` effectively *is* set when `wat` is null(denoting "all"); which suggests it might not be entirely useless in the API proper.